pkg/ccl/changefeedccl/sink_external_connection.go

```diff
@@ -0,0 +1,64 @@
+// Copyright 2022 The Cockroach Authors.
+//
+// Licensed as a CockroachDB Enterprise file under the Cockroach Community
+// License (the "License"); you may not use this file except in compliance with
+// the License. You may obtain a copy of the License at
+//
+//     https://github.com/cockroachdb/cockroach/blob/master/licenses/CCL.txt
+
+package changefeedccl
+
+import (
+	"context"
+
+	"github.com/cockroachdb/cockroach/pkg/cloud/externalconn"
+	"github.com/cockroachdb/cockroach/pkg/cloud/externalconn/connectionpb"
+	"github.com/cockroachdb/cockroach/pkg/jobs/jobspb"
+	"github.com/cockroachdb/cockroach/pkg/kv"
+	"github.com/cockroachdb/cockroach/pkg/security/username"
+	"github.com/cockroachdb/cockroach/pkg/sql/execinfra"
+	"github.com/cockroachdb/cockroach/pkg/sql/sqlutil"
+	"github.com/cockroachdb/errors"
+)
+
+func makeExternalConnectionSink(
+	ctx context.Context,
+	u sinkURL,
+	user username.SQLUsername,
+	db *kv.DB,
+	ie sqlutil.InternalExecutor,
+	serverCfg *execinfra.ServerConfig,
+	feedCfg jobspb.ChangefeedDetails,
```

Comment:
Can you add a TODO to replace direct `jobspb.ChangefeedDetails` with `ChangefeedConfig` -- I want to make sure that we don't accidentally save modified details back in the jobs table.  (/cc @HonoreDB )